test/e2e/service.go

```diff
@@ -1386,26 +1404,35 @@ var _ = framework.KubeDescribe("Services", func() {
 		jig.SanityCheckService(svc, v1.ServiceTypeLoadBalancer)
 		Expect(isInternalEndpoint(lbIngress)).To(BeFalse())
 
-		By("switiching back to interal type LoadBalancer, with static IP specified.")
-		internalStaticIP := "10.240.11.11"
-		svc = jig.UpdateServiceOrFail(namespace, serviceName, func(svc *v1.Service) {
-			svc.Spec.LoadBalancerIP = internalStaticIP
-			svc.ObjectMeta.Annotations[serviceAnnotationLoadBalancerInternal] = "true"
-		})
-		framework.Logf("Waiting up to %v for service %q to have an internal LoadBalancer", createTimeout, serviceName)
-		if pollErr := wait.PollImmediate(pollInterval, createTimeout, func() (bool, error) {
-			svc, err := jig.Client.Core().Services(namespace).Get(serviceName, metav1.GetOptions{})
-			if err != nil {
-				return false, err
+		// GCE cannot test a specific IP because the test may not own it. This will chagne when
+		// IP reservation is supported
+		if framework.ProviderIs("azure") {
+			By("switching back to interal type LoadBalancer, with static IP specified.")
+			internalStaticIP := "10.240.11.11"
+			svc = jig.UpdateServiceOrFail(namespace, serviceName, func(svc *v1.Service) {
+				svc.Spec.LoadBalancerIP = internalStaticIP
+				setLBToInternal(svc)
+			})
+			framework.Logf("Waiting up to %v for service %q to have an internal LoadBalancer", createTimeout, serviceName)
+			if pollErr := wait.PollImmediate(pollInterval, createTimeout, func() (bool, error) {
+				svc, err := jig.Client.Core().Services(namespace).Get(serviceName, metav1.GetOptions{})
+				if err != nil {
+					return false, err
+				}
+				lbIngress = &svc.Status.LoadBalancer.Ingress[0]
+				return isInternalEndpoint(lbIngress), nil
+			}); pollErr != nil {
+				framework.Failf("Loadbalancer IP not changed to internal.")
 			}
-			lbIngress = &svc.Status.LoadBalancer.Ingress[0]
-			return isInternalEndpoint(lbIngress), nil
-		}); pollErr != nil {
-			framework.Failf("Loadbalancer IP not changed to internal.")
+			// should have the given static internal IP.
+			jig.SanityCheckService(svc, v1.ServiceTypeLoadBalancer)
+			Expect(framework.GetIngressPoint(lbIngress)).To(Equal(internalStaticIP))
+		}
+
+		if framework.ProviderIs("gce", "gke") {
+			By("switching to ClusterIP type to destroy loadbalancer")
+			jig.ChangeServiceType(svc.Namespace, svc.Name, v1.ServiceTypeClusterIP, createTimeout)
```

Comment:
Could remove this cloud-specific check. Didn't want to make functional changes to azure without permission from somebody.